### ISSUE TITLE
index_fill, index_copy, etc wrong parameter number/first parameter, causing issues in PyCharm

### ISSUE THREAD

**Issue by BlackHC** [pytorch/pytorch#55737]
## 📚 Documentation

tl;dr: Remove `tensor1` from the documentation of `index_fill`, `index_copy`, `index_put` and `index_add`.

It is confusing to readers. The downstream issue is that e.g.PyCharm infers the parameters for these methods from the documentation and creates spurious warnings. 

The documentation says:

```
>>> help(a.index_fill)

index_fill(...) method of torch.Tensor instance

index_fill(tensor1, dim, index, value) -> Tensor
    
    Out-of-place version of :meth:`torch.Tensor.index_fill_`.
    `tensor1` corresponds to `self` in :meth:`torch.Tensor.index_fill_`.
```

However, `tensor1` actually takes the place of `self`. You do not pass `tensor1` explicitly.

Example code:
```
x = torch.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]], dtype=torch.float)
index = torch.tensor([0, 2])
y = x.index_fill(1, index, -1)
```
Note, no `tensor1` here.

Compare to `index_fill_`:
```
index_fill_(...) method of torch.Tensor instance
    index_fill_(dim, index, val) -> Tensor
    
    Fills the elements of the :attr:`self` tensor with value :attr:`val` by
    selecting the indices in the order given in :attr:`index`.
```
which does not mention `self` and leads to the same call signature:
```
x = torch.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]], dtype=torch.float)
index = torch.tensor([0, 2])
x.index_fill_(1, index, -1)
```

Method docs generally do not mention `self`. Even though, the parameter is called `tensor1` for `index_fill` and so on, it takes the place of `self` and is not implicitly passed. Maybe this is an issue of how the documentation is generated.

PyCharm creates spurious warnings based on this: "Parameter 'tensor1' unfilled".

It also took me a while to understand the documentation because I was wondering: if I pass `tensor1`, why am I calling this as a method when it ignores `self`. Shouldn't this be a static method then or part of the torch namespace?

Thanks!
 

cc @brianjo @mruberry

**Comment by ngimel** [comment]
Thanks for reporting, we'll accept a PR fixing this

**Comment by bdhirsh** [comment]
reserving for onboarding

**Comment by NivekT** [comment]
Fixed by #60486
